src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentStack.cs

```diff
@@ -410,7 +410,7 @@ private static void ValidatePushPopRangeInput(T[] items, int startIndex, int cou
                 throw new ArgumentOutOfRangeException(nameof(count), SR.ConcurrentStack_PushPopRange_CountOutOfRange);
             }
             int length = items.Length;
-            if (startIndex >= length || startIndex < 0)
```

Comment:
Doesn't this change the exception observed when `startIndex == length`, with a non-zero `count`?  You've added a test that catches `ArgumentException` for this case - is that the exception that occurs before this change?  (Also, it may be good to add a similar case for a non-empty array regardless)

Not that anybody should be catching `ArgumentException`, so as breaking changes go this is relatively minor...